### ISSUE TITLE
Server reboot issues - reports not populating

### ISSUE THREAD

**Issue by kdflint** [kdflint/speedybooks-backlog#18]
Make sure when server is being rebooted or started the user data doesn’t get wiped off

**Comment by kdflint** [comment]
Diagnosis: When the node.js server application that serves the api calls is not running, then the reports on the web site return no data. Since this server was not configured as a service, then it shut down anytime the server was rebooted. Upon restart, the reports were not serving until the api service was started up again manually with this command: `:$ sudo nohup node \srv\fusion-speedybooks\app.js &` 

In addition, even when running, the node.js application was not world-reachable as it listens on port 1337 which is correctly closed behind the firewall. The api calls from the browser were specific to port 1337 so never getting through. The telltale symptom that the api calls are not reaching their endpoint are that the web page "hangs" and eventually times out. 

The error reported in the javascript browser console for both of the above problems is

Error: rejection.data.message is undefined
_responseError@https://app.speedybooks.net/scripts/authInterceptorService.js:28:1
x@https://app.speedybooks.net/scripts/vendor.js:106:206
x@https://app.speedybooks.net/scripts/vendor.js:106:206
x@https://app.speedybooks.net/scripts/vendor.js:106:206
then/<@https://app.speedybooks.net/scripts/vendor.js:107:468
$eval@https://app.speedybooks.net/scripts/vendor.js:118:28
$digest@https://app.speedybooks.net/scripts/vendor.js:115:113
$apply@https://app.speedybooks.net/scripts/vendor.js:118:360
h@https://app.speedybooks.net/scripts/vendor.js:78:325
x@https://app.speedybooks.net/scripts/vendor.js:83:288
we/</z.onreadystatechange@https://app.speedybooks.net/scripts/vendor.js:84:340

**Comment by kdflint** [comment]
Two things were solved.

First. we set up a reverse proxy with nginx so that the api call was not directed to port 1337 but rather to the default https port 443. Nginx receives the call and now forwards the request to the node service running behind the firewall.

Previously the api call from the web page looked like `http://fusion.speedybooks.net:1337`  
Now, proxied and secured, it looks like: `https://fusion.speedybooks.net?`

Second, we installed and configured PM2, a process manager, to provide a process interface and service registration. Now on reboot the api service will resume with no manual intervention. Also we have lots of nifty process insights to check the health of the process in real time.

Outstanding cheat sheet for both the above can be found [here](https://www.digitalocean.com/community/tutorials/how-to-set-up-a-node-js-application-for-production-on-ubuntu-16-04)

Finally, note that PM2 is installed and invoked under user `kdflint`, so that `ps -ef` will show `node /srv/fusion-speedybooks/app.js` will be running under user `kdflint`
